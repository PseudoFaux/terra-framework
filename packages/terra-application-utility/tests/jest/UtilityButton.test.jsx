import React from 'react';
import UtilityButton from '../../src/utility/_UtilityButton';
import MockConfig from '../../examples/index-examples/MockConfig';

describe('UtilityButton', () => {
  const mockOnChange = jest.fn();
  const mockOnDisclose = jest.fn();
  const mockOnRequestClose = jest.fn();
  const accessory = <Image />;
  const title = 'name';
  const variant = 'header';

  it('should render with default props', () => {
    const wrapper = shallow(
      <UtilityButton
        menuItems={[]}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
        variant={variant}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a title', () => {
    const wrapper = shallow(
      <UtilityButton
        menuItems={[]}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
        title={title}
        variant={variant}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with an accessory', () => {
    const wrapper = shallow(
      <UtilityButton
        menuItems={[]}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
        accessory={accessory}
        variant={variant}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with menu items', () => {
    const wrapper = shallow(
      <UtilityButton
        menuItems={MockConfig()}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
        variant={variant}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('it should pass in a custom prop', () => {
    const wrapper = shallow(
      <UtilityButton
        menuItems={[]}
        onChange={mockOnChange}
        onDisclose={mockOnDisclose}
        onRequestClose={mockOnRequestClose}
        variant={variant}
        id="test"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
