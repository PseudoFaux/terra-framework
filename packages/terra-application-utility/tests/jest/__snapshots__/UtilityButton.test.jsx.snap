// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityButton it should pass in a custom prop 1`] = `
<button
  aria-label="Utility Button"
  className="header-utility-button"
  id="test"
  onClick={[Function]}
  onRequestClose={[Function]}
>
  <IconChevronDown
    className="icon"
  />
</button>
`;

exports[`UtilityButton should render with a title 1`] = `
<button
  aria-label="Utility Button"
  className="header-utility-button"
  onClick={[Function]}
  onRequestClose={[Function]}
>
  <span
    className="title"
  >
    name
  </span>
  <IconChevronDown
    className="icon"
  />
</button>
`;

exports[`UtilityButton should render with an accessory 1`] = `
<button
  aria-label="Utility Button"
  className="header-utility-button"
  onClick={[Function]}
  onRequestClose={[Function]}
>
  <Image
    className="accessory"
  />
  <IconChevronDown
    className="icon"
  />
</button>
`;

exports[`UtilityButton should render with default props 1`] = `
<button
  aria-label="Utility Button"
  className="header-utility-button"
  onClick={[Function]}
  onRequestClose={[Function]}
>
  <IconChevronDown
    className="icon"
  />
</button>
`;

exports[`UtilityButton should render with menu items 1`] = `
<button
  aria-label="Utility Button"
  className="header-utility-button"
  onClick={[Function]}
  onRequestClose={[Function]}
>
  <IconChevronDown
    className="icon"
  />
</button>
`;
